dts/arm/st/h7/stm32h745.dtsi

```diff
@@ -27,21 +27,69 @@
 	* (RM0399 Rev 3)
 	*/
 
-	/* System data RAM accessible over AXI bus: AXI SRAM in D1 domain */
 	sram0: memory@24000000 {
-		reg = <0x24000000 DT_SIZE_K(512)>;
+		reg = <0x24000000 DT_SIZE_K(864)>;
 		compatible = "mmio-sram";
-	};
 
-	/* System data RAM accessible over AHB bus: SRAM1 in D2 domain */
-	sram1: memory@30000000 {
-		reg = <0x30000000 DT_SIZE_K(288)>;
-		compatible = "mmio-sram";
-	};
+		#address-cells = <1>;
+		#size-cells = <1>;
 
-	/* System data RAM accessible over AHB bus: SRAM4 in D2 domain  */
-	sram4: memory@38000000 {
-		reg = <0x38000000 DT_SIZE_K(64)>;
-		compatible = "mmio-sram";
+		ranges = <
```

Comment:
`mmio-sram` compatible does not specify `ranges` as a property nor child bindings (here we have a parent and a child with the same compatible). I think this needs more discussion as well as proper support in Zephyr, not just definitions.

Reply:
> `mmio-sram` compatible does not specify `ranges` as a property nor child bindings (here we have a parent and a child with the same compatible). I think this needs more discussion as well as proper support in Zephyr, not just definitions.

Where can it be better discussed?

Reply:
@Nukersson #dev-review meeting

Reply:
Is there a need to have these as one node v separate nodes for each SRAM ?

Reply:
> Is there a need to have these as one node v separate nodes for each SRAM ?

With this approach I am able to use all SRAM sections of H7 together.

Reply:
My initial feeling is that I'd suggest that be done as a board specific setting of CONFIG_SRAM_SIZE/CONFIG_SRAM_BASE_ADDRESS.  I'm concerned about the precedence set of describing multiple SRAM regions this way and what assumptions one can really make.

What are you pointing `zephyr,sram` at in the case you are using all the SRAM sections together?

Reply:
> My initial feeling is that I'd suggest that be done as a board specific setting of CONFIG_SRAM_SIZE/CONFIG_SRAM_BASE_ADDRESS. I'm concerned about the precedence set of describing multiple SRAM regions this way and what assumptions one can really make.
> 
> What are you pointing `zephyr,sram` at in the case you are using all the SRAM sections together?

Compiling `samples/hello_world` with `zephyr,sram = &axi_sram;` gives:
```
[129/136] Linking C executable zephyr/zephyr_prebuilt.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       16580 B         1 MB      1.58%
            DTCM:          0 GB       128 KB      0.00%
            SRAM:        4512 B       512 KB      0.86%
        IDT_LIST:         120 B         2 KB      5.86%
[136/136] Linking C executable zephyr/zephyr.elf
```

and with `zephyr,sram = &sram0;` gives:
```
[129/136] Linking C executable zephyr/zephyr_prebuilt.elf
Memory region         Used Size  Region Size  %age Used
           FLASH:       16580 B         1 MB      1.58%
            DTCM:          0 GB       128 KB      0.00%
            SRAM:        4512 B       864 KB      0.51%
        IDT_LIST:         120 B         2 KB      5.86%
[136/136] Linking C executable zephyr/zephyr.elf
```

Especially see difference in `Region Size` of `SRAM` section.

Thus my approach still allows using of each separate SRAM section within the whole SRAM.

Reply:
Thanks.  

I still see this as an application specific need and not a generic solution.  It just happens that all the SRAM regions are contiguous so things work out that way.

In addition, the way you are using `ranges` + `reg` is technically in violation of DTS.  If you have a `range` the node that has that shouldn't have a `reg` that overlaps that `range` region.  Since you are effectively claiming 2 different device nodes for the same `reg` region than.

Reply:
Ok. I see your point. Thank you very much. But how would you put different memory banks of some SoC with non-consequtive memory addresses together? Would you use device-tree or linker skript?